bundles/org.openhab.binding.resol/README.md

```diff
@@ -0,0 +1,196 @@
+ * Sonnenkraft SKSC3+
+ * Sonnenkraft SKSC3HE
+ * Sonnenkraft SKSR 1/2/3
+ * Vitosolic 200
+ * COSMO Multi
+ * Drainback DeDietrich
+ * Diemasol C
+
+A more complete list can be found in the doc of the [resol-vbus-java library](http://danielwippermann.github.io/resol-vbus/vbus-packets.html).
+
+## Discovery
+
+Discovery is tested for VBus-LAN adapters DL2, DL3 and KM2 devices, it should also work for other devices providing a live data port.
+After a bridge is detected in the local network the password needs to be given and the things on the VBUS will popup in the inbox.
+
+## Binding Configuration
+
+The Resol binding doesn't need any form of configuration in files.
+
+## Bridge Configuration
+
+The bridge is the device connecting the Resol VBUS to the network, usually a VBus-LAN adapter or integrated in some of the solar controllers like DL3.
+For the connection from the Resol binding the bridge requires the configuration of
+
+| property             | type    | Required | description                                                |
+|----------------------|---------|----------|------------------------------------------------------------|
+| ipAddress            | String  | yes      | IP address or hostname of the VBUS adapter                 |
+| password             | String  | yes      | Password, defaults to 'vbus' for factory setting devices   |
+| port                 | Number  | no       | Port for the TCP connection, defaults to 7053              |
+| adapterSerial        | String  | no       | Serialnumber of the device (informative only)              |
+
+
+## Thing Configuration
+
+Depending on the solar/heating controller you have attached to your VBUS there will be a "controller" and several other things like heat quantity meters, heating circuit controls, etc.
+These do not require any configuration parameters and will pop up in your inbox after the bridge has received data from them.
+The name of the devices is usually the Resol name with spaced replaced by _ and a "-Controller" suffix like "DeltaSol_MX-Controller".
+For configuration in files you can enable the logging with at least INFO level for the resol binding and search the logs for "ThingHandler for (.*) not registered." to identify the names of the things you can add for your VBUS devices.
+
+
+## Channels
+
+The channels of a thing are determined automatically based on the received VBUS data and are highly dependent on the used device.
+Here is a list of the channels of a DeltaSol MX with a heat quantity meter (HQM) and an extension module EM.
+The channels supported for your device can be seen after autodiscovery or 
+
+| channel                       | type                     | description                                        |
+|-------------------------------|--------------------------|----------------------------------------------------|
+| Pump_speed_relay_x            | Number:Dimensionless     | Percentage of the output state of relay 'x'        |
```

Comment:
> Thanks for your contribution again! I reviewed your code and here is my feedback.

Thanks a lot. Some changes will be quite easy, others take some time but I'll tackle them in the upcoming days.

> What is the specification file with the .vsf extension about?
The used library vbus-java is using this internal file as the definition of communication on the resol VBUS. It is included in the library jar but for whatever reason the lib doesn't find it in it's own distribution when integrated into the openHAB build system. Maybe it automagically works now without since the library is included via mvn instead of copying the jar or you could give me some advice how to configure a resource search path to integrate the jar of the library to locate this file?

Reply:
ok. I use the name as defined by Resol and output by the vbus-java library and replace spaces and some other characters which are not allowed as channel names. Should I make them all lower case or just the first char?

Reply:
If you use underscores to separate the words, all letters should be lower case.

Reply:
the underscores are replacements of spaces from the name in the library. But you're fully right it doesn't look good and I will just make all channel names lower case.